src/app/(protected)/[group]/[subGroup]/[instance]/(admin-panel)/(stage-6)/export-data/_components/export-button.tsx

```diff
@@ -0,0 +1,63 @@
+"use client";
+import { DownloadIcon } from "lucide-react";
+import { unparse } from "papaparse";
+import { useEffect, useRef } from "react";
+
+import { Button } from "@/components/ui/button";
+
+import { AllocationCsvData } from "@/lib/validations/allocation-csv-data";
+
+export function ExportDataButton({ data }: { data: AllocationCsvData[] }) {
+  const downloadLinkRef = useRef<HTMLAnchorElement>(null);
+
+  const header = [
+    "Project Internal ID",
+    "Project External ID",
+    "Student ID",
+    "Project Title",
+    "Project Description",
+    "Project Special Technical Requirements",
+    "Student Ranking",
+  ];
+
+  const rows = data.map((e) => [
+    e.projectInternalId,
+    e.projectExternalId,
+    e.studentId,
+    e.projectTitle,
+    e.projectDescription,
+    e.projectSpecialTechnicalRequirements,
+    e.studentRanking,
+  ]);
+
+  const csvContent = unparse({
+    fields: header,
+    data: rows,
+  });
+
+  useEffect(() => {
+    if (downloadLinkRef.current) {
+      const link = downloadLinkRef.current;
+      link.href = encodeURI(`data:text/csv;charset=utf-8,${csvContent}`);
+      link.download = "candidate_data.csv";
```

Comment:
Not sure on naming here. I think project_allocation_data or something similar may be better.

Reply:
Oh yeah, that actually makes no sense. I'm happy with `project_allocation_data.csv`